Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -255,89 +261,26 @@ void UPSWorldSubsystem::AddProgressionStarActors()
 void UPSWorldSubsystem::OnTakeActorsFromPoolCompleted(const TArray<FPoolObjectData>& CreatedObjects)
 {
 	const FPSRowData& CurrentRowData = GetCurrentRow();
+	float AmountUnlocked = CurrentRowData.CurrentLevelProgression;
+	FVector PreviousActorLocation;
 
-	float CurrentAmountOfUnlocked = 0;
-	float CurrentAmountOfLocked = 0;
-
-	//set updated amount of stars
-	if (CurrentRowData.CurrentLevelProgression >= CurrentRowData.PointsToUnlock)
-	{
-		// set required points (stars)  to achieve for a level  
-		CurrentAmountOfUnlocked = CurrentRowData.PointsToUnlock;
-		CurrentAmountOfLocked = 0;
-	}
-	else
-	{
-		// Calculate the unlocked against locked points (stars) 
-		CurrentAmountOfUnlocked = CurrentRowData.CurrentLevelProgression;
-		CurrentAmountOfLocked = CurrentRowData.PointsToUnlock - CurrentRowData.CurrentLevelProgression;
-	}
-
-	float integerPart;
-	float fractionalPart;
-	// Setup spawned widget
 	for (const FPoolObjectData& CreatedObject : CreatedObjects)
 	{
-		if (CurrentAmountOfUnlocked > 0)
-		{
-			fractionalPart = modff(CurrentAmountOfUnlocked, &integerPart);
-			// check if it has a fractional part and if there is no more fully achieved stars 
-			if (fractionalPart < 1.0f && CurrentAmountOfUnlocked < 1.0f)
-			{
-				UpdateStarActor(CreatedObject, fractionalPart, 0);
-			}
-			else
-			{
-				UpdateStarActor(CreatedObject, 1, 0);
-			}
-
-			CurrentAmountOfUnlocked--;
-			continue;
-		}
-
-		if (CurrentAmountOfLocked > 0)
-		{
-			UpdateStarActor(CreatedObject, 0, 1);
-			CurrentAmountOfLocked--;
-		}
-	}
-}
-
-// Updates star actor to locked/unlocked according to input amounnt
-void UPSWorldSubsystem::UpdateStarActor(const FPoolObjectData& CreatedData, float AmountOfUnlockedStars, float AmountOfLockedStars)
-{
-	AActor& SpawnedActor = CreatedData.GetChecked<AActor>();
-	UStaticMeshComponent* MeshComponent = SpawnedActor.FindComponentByClass<UStaticMeshComponent>();
-
-	const FPSRowData& CurrentRowData = GetCurrentRow();
-
-	SpawnedStarActorsInternal.Add(&SpawnedActor);
-
-	SpawnedActor.SetActorTransform(CurrentRowData.StarActorTransform);
-	// if the actor is the first element it set initial position
-	// from the initial position there is a distance between stars 
-	if (SpawnedStarActorsInternal.Num() > 1)
-	{
-		SpawnedActor.SetActorLocation(SpawnedStarActorsInternal[SpawnedStarActorsInternal.Num() - 2]->GetActorLocation() + CurrentRowData.OffsetBetweenStarActors);
-	}
+		APSStarActor& SpawnedActor = CreatedObject.GetChecked<APSStarActor>();
 
-	if (AmountOfUnlockedStars > 0) //unlocked stars
-	{
-		if (AmountOfUnlockedStars > 0 && AmountOfUnlockedStars < 1) // dynamic 
+		float StarAmount = FMath::Clamp(AmountUnlocked, 0.0f, 1.0f);
+		if (AmountUnlocked > 0)
 		{
-			MeshComponent->SetMaterial(0, StarDynamicProgressMaterial);
-			StarDynamicProgressMaterial->SetScalarParameterValue(TEXT("Percentage2"), AmountOfUnlockedStars / 3);
+			SpawnedActor.UpdateStarActorMeshMaterial(StarDynamicProgressMaterial, StarAmount, false);
```

Comment:
Please refactor the `UpdateStarActorMeshMaterial` function to avoid using raw boolean arguments. You can either introduce an enum for clarity or use a named boolean constant in your function calls.

### Why the change is needed:
- Raw boolean values like `true` or `false` do not convey meaning, making the code harder to understand. Using an enum or a named constant makes the purpose of the argument clear to readers and maintainers.

Option A: Use an enum for clarity.

```cpp
// Define an enum for star state
UENUM()
enum class EPSStarState : uint8
{
    None,
    Locked,
    Unlocked
};

// Refactor the function signature to accept an enum
void UpdateStarActorMeshMaterial(UStaticMesh* Mesh, float StarAmount, EPSStarState StarState);

// Update function calls with the new enum
SpawnedActor.UpdateStarActorMeshMaterial(StarDynamicProgressMaterial, StarAmount, EPSStarState::Unlocked);
```

Alternative option B (less preferable): Use a named constant for the boolean.

```cpp
// Refactor with a named constant for clarity
constexpr bool bShouldLock = false;

// Update the function call with a named boolean
SpawnedActor.UpdateStarActorMeshMaterial(StarDynamicProgressMaterial, StarAmount, bShouldLock);
```

Reply:
I went to option B: https://github.com/h4rdmol/ProgressionSystem/pull/17/commits/1469e2393bcd04796da5e3dae5d4ee9b149ebe9e
After resolving the conflicts I will take to modify the enum